deepchecks/checks/performance/class_performance_imbalance.py

```diff
@@ -0,0 +1,239 @@
+                    self.alternative_metrics[name] = get_scorer(metric)
+                else:
+                    raise ValueError(
+                        f"alternative_metrics - expected to receive 'Mapping[str, Callable]' but got " #pylint: disable=inconsistent-quotes
+                        f"'Mapping[str, {type(metric).__name__}]'!" #pylint: disable=inconsistent-quotes
+                    )
+
+    def run(
+        self,
+        dataset: Dataset,
+        model: t.Any # TODO: find more precise type for model
+    ) -> CheckResult:
+        """Run Check.
+
+        Args:
+            dataset (Dataset): a dataset object
+            model (BaseEstimator): A scikit-learn-compatible fitted estimator instance
+
+        Returns:
+            CheckResult: value is a dictionary of labels, value of each key is another dictionary
+                with calculated metrics
+
+        Raises:
+            DeepchecksValueError:
+                if task type is not binary or multi-class;
+                if dataset does not have a label column or the label dtype is unsupported;
+                if provided dataset is empty;
+        """
+        return self._class_performance_imbalance(dataset, model)
+
+    def _class_performance_imbalance(
+        self,
+        dataset: Dataset,
+        model: t.Any # TODO: find more precise type for model
+    ) -> CheckResult:
+        check_name = type(self).__name__
+        expected_model_types = [ModelType.BINARY, ModelType.MULTICLASS]
+
+        dataset.validate_label(check_name)
+        dataset.validate_features(check_name)
+        Dataset.validate_dataset(dataset, check_name)
+        task_type_validation(model, dataset, expected_model_types, check_name)
+
+        labels = t.cast(pd.Series, dataset.label_col())
+        features = t.cast(pd.DataFrame, dataset.features_columns())
+
+        if self.alternative_metrics is not None:
+            df = pd.DataFrame.from_dict(self._execute_alternative_metrics(
+                model, features, labels
+            ))
+        else:
+            y_true = labels
+            y_pred = model.predict(features)
+            unique_labels = get_unique_labels(labels, y_pred)
+            df = pd.DataFrame.from_dict({
+                name: dict(zip(unique_labels, metric_func(y_true, y_pred, labels=unique_labels))) # type: ignore
+                for name, metric_func in self._get_default_metrics().items()
+            })
+
+        def display():
+            title = (
+                'Class Performance Imbalance Check for binary data'
+                if len(labels) == 2
+                else 'Class Performance Imbalance Check for multi-class data'
+            )
+
+            df.transpose().plot.bar(
+                title=title,
+                backend='matplotlib',
+                xlabel='Metrics',
+                ylabel='Values'
+            )
+
+        return CheckResult(
+            value=df.transpose().to_dict(),
```

Comment:
why not instead of dict -> df -> dict, just get the dict and only in the display function convert it to dataframe?

Reply:
from another reply:

> we want the value to be in the format {'label_name': {'metric_name': value}}, without DataFrame we would need to 
> make some extra work to achieve this.
>
> What I mean is that with help of DataFrame.transpose implementation looks cleaner and more readable.